### PR TITLE
245: replace deprecated DataFrame.applymap with DataFrame.map

### LINKED ISSUE #245: [Issue]: Multiple files use deprecated DataFrame.applymap which should be replaced with DataFrame.map
### Environment
- QF-Lib: 4.0.4
- Python: 3.11.9

### What happened?
When running scripts that generate tearsheets or use backtest monitoring, FutureWarnings are displayed:

```
qf-lib\qf_lib\analysis\tearsheets\portfolio_analysis_sheet.py:141: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.                                                                               
  positions_history = positions_history.applymap(

qf-lib\qf_lib\analysis\tearsheets\portfolio_analysis_sheet.py:213: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.                                                                               
  assets_history = assets_history.applymap(

qf-lib\qf_lib\analysis\tearsheets\portfolio_analysis_sheet.py:276: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.                                                                               
  positions[["Long", "Short", "Out"]] = positions[["Long", "Short", "Out"]].applymap(lambda x: '{:.2%}'.format(x))

qf-lib\qf_lib\analysis\tearsheets\portfolio_analysis_sheet.py:319: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.                                                                               
  applymap(lambda pnl_series: pnl_series.iloc[-1] if not pnl_series.empty else 0.0). \

qf-lib\qf_lib\analysis\tearsheets\portfolio_analysis_sheet.py:321: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.                                                                               
  applymap(lambda p: '{:,.2f}'.format(p)). \

qf-lib\qf_lib\backtesting\monitoring\backtest_monitor.py:334: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.    
  df = signals_df.applymap(fun)
```

The `applymap` method is deprecated in pandas and should be replaced with `map`.

### PR DESCRIPTION
**Pull request by sneakyjbras**
Closes #245 

**Disclaimer**: This is my first pull request ever for this repo. I might be missing the point. Also, considering it is my first pull request I have added extensive unit testing, covering as many cases and corner cases as possible. 

**Bulk of pull request**:

`DataFrame.applymap` was deprecated in pandas 2.1.0 and removed in pandas 3.0.
The fix is straightforward: replace every `.applymap(` call with `.map(`.

14 call sites across 5 files:

- `analysis/tearsheets/portfolio_analysis_sheet.py` — 5
- `analysis/model_params_estimation/model_params_evaluator.py` — 5
- `backtesting/monitoring/backtest_monitor.py` — 1
- `data_providers/bloomberg_dl/bloomberg_dl_data_provider.py` — 1
- `analysis/strategy_monitoring/assets_monitoring_sheet.py` — 2
- `containers/dataframe/qf_dataframe.py` — docstring only

Added 24 unit tests in `tests/unit_tests/containers/test_dataframe_map_deprecation.py`.
Tests cover every lambda pattern used across the affected files and assert no
`FutureWarning` about `applymap` is raised.

**Side finding:** `QFSeries.__init__` passes `fastpath` to `pd.Series.__init__`,
which was removed in pandas 3.0. Unrelated to this fix, but worth tracking if
pandas 3.x support is ever needed.

**PS**: I tried keeping the logic as clean as possible. I used an assistant to digest me tests, based on your coding style and find corner cases that I could think of.

Cheers

### CODE REVIEW

File: run_issue_245_fix.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
```
Comment (myrmarachne) [inline run_issue_245_fix.sh]: I guess this file is safe to be deleted

File: qf_lib/tests/unit_tests/containers/test_dataframe_map_deprecation.py
Comment (myrmarachne) [inline qf_lib/tests/unit_tests/containers/test_dataframe_map_deprecation.py:1]: These tests mostly test the map function on manually designed pandas DataFrames and do not really check how the map / applymap is used by qf-lib, so I would say most of them are not necessary nor relevant for the library. The tests don't prove we are not getting the warning within the usual flow or that the code works as expected, but we have other tests already doing this, so considering what the fix is, I think we can safely remove this file.

**Review by myrmarachne — CHANGES_REQUESTED** [review]
Hello and thank you for your first pull request to the repo! We really appreciate it, together with the heads-up regarding the Pandas 3 fastpath issue.

I reviewed the code and ran the tests locally, including integration tests for our private data providers. Everything passes and all applymap usages are removed.

To get this ready to merge, we just need a few small adjustments:

- During the review, I noticed two files that might actually be unnecessary for this fix. Have a look and see if we can remove them.
- The GitHub pipeline failed the style check. You can see the errors in the action output, but it’s usually easiest to run flake8 locally to catch and fix them.

Once the formatting is cleaned up we will be ready to merge it. Thanks again for your first contribution!

File: run_issue_245_fix.sh (hunk 1 above)
Comment (sneakyjbras) [reply run_issue_245_fix.sh]: Hello, yes indeed. I included this file (generated by my agent) so you guys could reproduce my environment and the flow of execution I had. The fix -> running tests I included. Will delete it from the PR.

File: qf_lib/tests/unit_tests/containers/test_dataframe_map_deprecation.py
Comment (sneakyjbras) [reply qf_lib/tests/unit_tests/containers/test_dataframe_map_deprecation.py:1]: Okay, thank you. Will remove.

**Comment by sneakyjbras** [comment]
Hello @myrmarachne I have removed the tests which I wrote to test my changes. And the script generated by my agent to reproduce my workflow / dev env. Sorry for the delay.

**Review by myrmarachne — APPROVED** [review]

**Review by ydelipinar — APPROVED** [review]
